patterns/vsi-extension/variables.tf

```diff
@@ -100,11 +100,13 @@ variable "block_storage_volumes" {
   description = "The list of block storage volumes to attach to each VSI. [Learn more](https://cloud.ibm.com/docs/vpc?topic=vpc-creating-block-storage&interface=ui#create-from-vsi)."
   type = list(
     object({
-      name           = string
-      profile        = string
-      capacity       = optional(number)
-      iops           = optional(number)
-      encryption_key = optional(string)
+      name              = string
+      profile           = string
+      capacity          = optional(number)
+      iops              = optional(number)
+      encryption_key    = optional(string)
+      resource_group_id = optional(string)
+      snapshot_id       = optional(string) # set if you would like to base volume on a snapshot
```

Comment:
I don't think we should expose this feature yet in landing zone (the snapshot recover) as it has not been fully tested. Also when we do adopt this feature, we should also have same feature on the boot volume, and perhaps consistency group option.

For now I would just remove snapshot_id option, and when we do add it we will probably focus on that feature for that PR and release.